src/data/movements.ts

```diff
@@ -1,30 +1,51 @@
 import { MovementAction } from './enums/MovementAction';
 
-import { Movement } from '../types/Actor';
+import { Actor, Movement } from '../types/Actor';
 
 import { isGroundCell } from '../utils/isGroundCell';
+import { Cell } from '../types/Cell';
 
 export const movements = {
   random: (self, playBoard) => {
-    const availableDirections = [];
+    const availableDirections: MovementAction[] = [];
 
-    if (isGroundCell(self.position.x - 1, self.position.y, playBoard)) {
-      availableDirections.push(MovementAction.Left);
-    }
-    if (isGroundCell(self.position.x, self.position.y - 1, playBoard)) {
-      availableDirections.push(MovementAction.Up);
-    }
-    if (isGroundCell(self.position.x + 1, self.position.y, playBoard)) {
-      availableDirections.push(MovementAction.Right);
-    }
-    if (isGroundCell(self.position.x, self.position.y + 1, playBoard)) {
-      availableDirections.push(MovementAction.Down);
+    const basicDirections = [
+      {action: MovementAction.Left, point: [-1, 0]},
+      {action: MovementAction.Up, point: [0, -1]},
+      {action: MovementAction.Right, point: [1, 0]},
+      {action: MovementAction.Down, point: [0, 1]},
+    ];
```

Comment:
Directions are identical for all movement types, so we can move them out of this object for future use in other movement algorithms